iothub/service/src/Messaging/Models/PurgeMessageQueueResult.cs

```diff
@@ -10,6 +10,21 @@ namespace Microsoft.Azure.Devices
     /// </summary>
     public sealed class PurgeMessageQueueResult
     {
+        /// <summary>
+        /// Initializes an instance of this class. Provided for unit testing purposes only.
+        /// </summary>
+        /// <param name="deviceId">
+        /// The Id of the device whose messages are being purged.
+        /// </param>
+        /// <param name="totalMessagesPurged">
+        /// The total number of messages that were purged from the device's queue.
+        /// </param>
+        public PurgeMessageQueueResult(string deviceId, int totalMessagesPurged)
```

Comment:
This class is public with public setters on the properties. It does not need a public ctor in its current state.

Also, if you add a ctor with params but don't offer one without, it can't be used to deserialize.